libs/s25main/ai/random.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2005 - 2025 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "helpers/random.h"
+#include <random>
+
+namespace AI {
+
+std::minstd_rand& getRandomGenerator();
+
+template<typename T>
+T randomValue(T min = std::numeric_limits<T>::min(), T max = std::numeric_limits<T>::max())
```

Comment:
Maybe a `bool randomBool` and `size_t randomIndex` makes sense to simplify code using `rand() % 2 == 0` and `randomValue(0, xx.size() - 1)`

Or instead of `randomBool` a `bool random(unsigned total, unsigned chance=1) -> randomValue(0u,total - 1u) <= chance`.   
Not sure I got the names and values right but it should replace `rand() % 10 == 0` by `random(10)` and `rand() % 10 > 5` by `random(10, 5)` and similar. This is the most common use so for. I.e. a "chance in total" chance to do something. Needs an assertion that `total > 0` and `chance < total` (a 5 out of 5 chance is a certain event). Or just make those 2 cases return false and true respectively